### PR TITLE
Update revert button text and description

### LINKED ISSUE #48695: Fix confusing messages prompting switch to classic templates
### Describe the solution you'd like

In #44931 Classic templates were renamed to avoid confusion, however, confusing messages remain in place when we're offering users the option to revert to these templates.

For example. When editing the Single Product template, there is a CTA button that says:

> Revert to Classic Product Template

The description beneath says:

> The **Single Product** template doesn’t allow for reordering or customizing blocks, but might work better with your extensions
Screenshot 2024-06-20 at 10 21 55 AM

I believe this is a mistake, as the merchant is currently editing the Single Product template, which _does_ allow for customization. 

The message would best be updated to say:

> The **Classic Product Template** doesn’t allow for reordering or customizing blocks, but might work better with your extensions

Screenshot 2024-06-20 at 10 22 47 AM

These messages appear and should be corrected on all the templates that offer a swap back to the classic version:

- Products by Attribute
- Products by Tag
- Products by Category
- Product Search results
- Product Catalog
- Single Product
- Order Confirmation
- Page: Cart
- Page: Checkout

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by imanish003**
### Changes proposed in this Pull Request:

I have updated the text of the button from "Revert to Classic Product Template" to "Revert to Classic Template". Also, I have updated the description below the button to include "(Classic)" in then End of the template name. This will help in making the message more clear and consistent for following templates:
- Products by Attribute
- Products by Tag
- Products by Category
- Product Search results
- Product Catalog
- Single Product

For more context, you may check this slack discussion: p1718959820891299-slack-C02FL3X7KR6

Closes #48695 

### How to test the changes in this Pull Request:

1. Go to "Products by Attribute" template
2. As shown in screenshot below, verify that 
  - The button text is "Revert to Classic Template"  
  - The description is "The [name] (Classic) template doesn’t allow for reordering or customizing blocks, but might work better with your extensions"
    - Here [name] is the name of the template you are currently editing
image

3. Verify it for other templates as well
  - Products by Tag
  - Products by Category
  - Product Search results
  - Product Catalog
  - Single Product

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Fix confusing messages prompting switch to classic templates

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/templates/revert-button/index.tsx [hunk 1]
```diff
@@ -98,21 +98,22 @@ const RevertClassicTemplateButton = () => {
...
+									`The <strongText /> template doesn’t allow for reordering or customizing blocks, but might work better with your extensions.`,
 									'woocommerce'
 								),
 								{
 									strongText: (
 										<strong>
-											{ template?.record?.title
-												?.rendered ?? '' }
+											{ template?.record?.title?.rendered
+												? `Classic ${ template.record.title.rendered }`
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/assets/js/templates/revert-button/index.tsx]: I suggest we make it aligned with the recent rename (https://github.com/woocommerce/woocommerce/pull/44931), which was:

- WooCommerce Product Attribute Block -> Product Attribute (Classic)
- WooCommerce Product Grid Block -> Product Grid (Classic)
- etc.

```suggestion
												? `${ template.record.title.rendered } (Classic)`
```

**Review by kmanijak — CHANGES_REQUESTED** [review]
Thanks for taking care of it. I made a suggestion so we keep the convention with the recent rename :)

File: plugins/woocommerce-blocks/assets/js/templates/revert-button/index.tsx (hunk 1 above)
Comment (imanish003) [reply plugins/woocommerce-blocks/assets/js/templates/revert-button/index.tsx]: Makes sense. I have made the changes in 4b7972f. Thanks for the suggestion. I will update the PR description accordingly.

**Review by kmanijak — COMMENTED** [review]
I don't think Order Confirmation, Page: Cart and Page: Checkout offer reverting to classic template, there's no option in Inspector Controls. Please let me know if I'm missing something or remove them from testing steps.

Also, I know it's nitpicking, but could you center the button label? 😅 I know it's a detail but it's hard to accept 😂 
image

**Comment by imanish003** [comment]
> I don't think Order Confirmation, Page: Cart and Page: Checkout offer reverting to classic template, there's no option in Inspector Controls. Please let me know if I'm missing something or remove them from testing steps.

Interesting! For example, I can see it on the "Order Confirmation" template, as you can see in the attached screenshot and demo video. But it isn't visible after refreshing the page. I am not 100% sure where this inconsistency is coming from. There is probably an issue with the logic used to show this button in the Inspector Controls.

image

https://github.com/woocommerce/woocommerce/assets/16707866/bc2b648e-b522-4c9a-a17e-981b737e0007

I followed following steps to reproduce this issue:
1. Go to templates page
2. Use "Cmd + K" to search for "Order Confirmation" template
3. Click on "Order Confirmation" template
4. You will see the "Revert to Classic Template" button in the Inspector Controls

But, issue isn't reproducing consistently. Based on this, my assumption is that button shouldn't be visible on "Order Confirmation" template. 

@kmanijak Are you able to reproduce this issue? I am going to remove these templates from testing steps for this PR as it isn't in scope of this PR. But we can create a separate issue for this if we are able to reproduce this issue.

> Also, I know it's nitpicking, but could you center the button label? I know it's a detail but it's hard to accept.

Thanks for your attention to detail 🚀. It makes sense. Therefore, I have centered the button label as per your suggestion.

**Review by kmanijak — APPROVED** [review]
Thanks for the change and fixing the button centering! 😄

**Pull request merged**